opentelemetry-api/src/trace/tracer.rs

```diff
@@ -271,9 +271,6 @@ pub struct SpanBuilder {
 
     /// Span status
     pub status: Status,
-
-    /// Sampling result
-    pub sampling_result: Option<SamplingResult>,
```

Comment:
This is necessary to support pre-sampled tracing implementations (e.g. `tracing-opentelemetry`) which require sampling information to be known before spans are created.

Reply:
That should be handled by either the `Sampler` or the `SpanProcessor` used. Either of which can handle this.

Reply:
[This](https://github.com/tokio-rs/tracing/blob/master/tracing-opentelemetry/src/tracer.rs#L66-L109) is the implementation that will be affected by this change. We can consider supporting alternate implementations to facilitate the intended behavior there but breaking support for `tracing` entirely would not be an ideal choice.

Reply:
I guess it doesn't hurt to provide additional ability outside the API. Integration with `tracing` has been important to opentelemetry adoption in Rust community IMO.

Reply:
Yea that makes sense, so then in that case it should be simplified down to. and re add the sampling_result.

```
let sampling_decision = if let Some(sampling_result) = builder.sampling_result.take() {
    self.process_sampling_result(sampling_result, parent_cx)
} else {                                      
    self.make_sampling_decision(
        parent_cx,
        trace_id,        
        &builder.name,        
        &span_kind,        
        &attribute_options,        
        link_options.as_deref().unwrap_or(&[]),        
        provider.config(),        
        &self.instrumentation_lib,        
    )        
}    
```

Reply:
Yeah that looks good to me 👍